### PR TITLE
Use java.nio.file.Path in FileVirtualIRFiles

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

File: linker/jvm/src/main/scala/org/scalajs/linker/irio/FileVirtualIRFiles.scala [hunk 1]
```diff
@@ -17,64 +17,67 @@ import scala.concurrent._
...
 
 import org.scalajs.ir
 
 object FileScalaJSIRContainer {
-  def fromClasspath(classpath: Seq[File])(
+  def fromClasspath(classpath: Seq[Path])(
       implicit ec: ExecutionContext): Future[Seq[FileScalaJSIRContainer]] = {
     Future.traverse(classpath) { entry =>
-      if (!entry.exists)
+      if (!blocking(Files.exists(entry)))
```
[the remark this answers is not in the document]
Comment (gzm0) [reply linker/jvm/src/main/scala/org/scalajs/linker/irio/FileVirtualIRFiles.scala]: You mean inside the lambda to `traverse`? Looking at this again, it seems that `fromDirectory` should not even return a future...

I'll revise this. It's indeed not ideal.

[the remark this answers is not in the document]
Comment (sjrd) [reply linker/jvm/src/main/scala/org/scalajs/linker/irio/FileVirtualIRFiles.scala]: > You mean inside the lambda to traverse?

Yes, indeed.

**Comment by gzm0** [comment]
Updated. I managed to simplify even further.

**Comment by sjrd** [comment]
CI says that `MainGenericRunner.scala` in partest needs to be updated.

**Comment by gzm0** [comment]
What a surprise :P Fixed

**Pull request merged**